src/Controllers/GithubApiController.php

```diff
@@ -0,0 +1,167 @@
+    {
+    }
+
+    /**
+     * Tries to branch `$branch_name` and head it to `$from_branch`.
+     * Throws a RuntimeException if unable to do so.
+     *
+     * @param string $branch_name
+     * @param string $from_branch
+     *
+     * @throws RuntimeException
+     * @return void
+     */
+    public function createBranch(string $branch_name, string $from_branch): void
+    {
+        $ref_name = sprintf('refs/heads/%1$s', $branch_name);
+        $ref_data = [
+            'ref' => $ref_name,
+            'sha' => $this->getBranchSha($from_branch),
+        ];
+
+        $response = $this->client
+            ->api('gitData')
+            ->references()
+            ->create($this->repo_owner, $this->repo_name, $ref_data);
+
+        if (is_array($response) && $response['ref'] === $ref_name) {
+            return;
+        }
+
+        throw new RuntimeException(sprintf('Unable to create branch, GitHub response was: %1$s', $response));
+    }
+
+    /**
+     * Returns branch SHA
+     *
+     * @param string $branch_name
+     *
+     * @return string|null
+     */
+    public function getBranchSha(string $branch_name): ?string
+    {
+        try {
+            $reference = $this->client
+                ->api('gitData')
+                ->references()
+                ->show($this->repo_owner, $this->repo_name, sprintf('heads/%1$s', $branch_name));
+
+            return is_array($reference)
+                ? $reference['object']['sha'] ?? null
+                : null;
+        } catch (Exception $e) {
+            return null;
+        }
+    }
+
+    /**
+     * Returns selected branch/file SHA
+     *
+     * @param string $path_to_file
+     * @param string $branch
+     *
+     * @throws RuntimeException
+     * @return string
+     */
+    public function getFileSha(string $path_to_file, string $branch): string
+    {
+        $response = $this->client
+            ->api('repo')
+            ->contents()
+            ->show($this->repo_owner, $this->repo_name, $path_to_file, $branch);
+        $sha      = $response['sha'] ?? null;
+        if (is_string($sha)) {
+            return $sha;
+        }
+
+        throw new RuntimeException(sprintf('Unable to retrieve file SHA: %1$s', $path_to_file));
+    }
+
+    /**
+     * Retrieves remote repository's file content
+     *
+     * @param string $path_to_file Full path to the file
+     *
+     * @throws JsonException
+     * @throws RuntimeException
+     * @return string
+     */
+    public function getFileContent(string $path_to_file): string
+    {
+        $file_data = $this->client
+            ->api('repo')
+            ->contents()
+            ->show($this->repo_owner, $this->repo_name, $path_to_file);
+        if (!$file_data || !isset($file_data['content'])) {
+            throw new RuntimeException(sprintf('Missing "%1$s" content', $path_to_file));
+        }
+
+        return base64_decode($file_data['content']);
```

Comment:
1. how about setting the strict argument?

```php
base64_decode(string $string, bool $strict = false): string|false
```

2. also the code can crash since the result maybe be `false` and you only support `string` as a return type.

Reply:
fixed